src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java

```diff
@@ -51,18 +51,28 @@
     /** The initial current value. */
     private static final BigInteger INITIAL_VALUE = BigInteger.ONE;
 
-    /** Stack of values - all but the current value. */
-    private final Deque<BigInteger> valueStack = new ArrayDeque<>();
+    /**
+     * Stack of NP values for ranges.
+     */
+    private final Deque<BigInteger> rangeValues = new ArrayDeque<>();
+
+    /** Stack of NP values for expressions. */
+    private final Deque<Integer> expressionValues = new ArrayDeque<>();
```

Comment:
Because NPath value for one expression is small number, unlike NPath value for whole file that can be huge.